src/lib/run_environment.ml

```diff
@@ -94,10 +94,37 @@ module Tool = struct
   end
 end
 
+module Make_fun = struct
+  module Requirement = struct
+    type t = [
+      | `Processors of int
+      | `Internet_access
+      | `Memory of [ `GB of float | `Decent | `Big ]
```

Comment:
@rleonid  `dont_schedule` already makes weird assumptions (can the user actually "not" schedule? does schedule mean something their context?), the prevous API has `quick_run_program`, so `` `Quick_run`` replaced it.

@ihodes how would we put values for `` `GB _``? Right now I've been using the `` `Big`` and `` `Decent` because that's the only thing that has a tractable meaning.
Moreover using the combinators below make more sense `streamish` *is* “decent-memory requirement on one processor” (like `gunzip`, or `sam-to-bam`)